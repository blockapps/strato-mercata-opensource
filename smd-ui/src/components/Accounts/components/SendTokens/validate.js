import { isOauthEnabled } from "../../../../lib/checkMode";

export const validate = (values) => {
  const errors = {};

  if (!values.from) {
    errors.value = 'Please select a user';
  }

  if (!values.fromAddress) {
    errors.value = 'Please select a address';
  }

  if (isOauthEnabled() && !values.address) {
    errors.value = "Please enter address"
  }

  if (!isOauthEnabled()) {

    if (!values.password) {
      errors.value = 'Please enter a password';
    }
    if (!values.radio && !values.toAddress) {
      errors.value = "Please select address"
    }

    if (values.radio === "0" && !values.toAddress) {
      errors.value = "Please select address"
    }

    if (values.radio === "1" && !values.address) {
      errors.value = "Please enter address"
    }
  }

  if (!values.value) {
    errors.value = 'Please enter a value';
  }

  return errors;
};

export default validate;
