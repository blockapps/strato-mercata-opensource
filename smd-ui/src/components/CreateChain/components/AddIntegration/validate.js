export function validate(values) {
  const errors = {};

  return errors;
}