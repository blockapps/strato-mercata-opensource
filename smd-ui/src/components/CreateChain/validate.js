export function validate(values) {
  const errors = {};

  if (!values.chainName) {
    errors.chainName = 'required';
  }

  if (values.members && !values.members.length) {
    errors.members = 'at least add one member';
  }

  return errors;
}