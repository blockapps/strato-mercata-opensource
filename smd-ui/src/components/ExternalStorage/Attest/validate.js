import { isOauthEnabled } from "../../../lib/checkMode";

export default function validate(values) {
  const errors = {};

  if (!isOauthEnabled() && !values.username) {
    errors.username = 'Select username';
  }

  if (!isOauthEnabled() && !values.address) {
    errors.address = 'Address required';
  }

  if (!isOauthEnabled() && !values.password) {
    errors.password = 'Password required';
  }

  if (!values.contractAddress) {
    errors.contractAddress = "Can't be blank";
  }

  return errors;
}