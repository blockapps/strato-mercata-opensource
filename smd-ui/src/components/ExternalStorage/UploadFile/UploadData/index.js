import React, { Component } from 'react';
import { Button, Intent } from '@blueprintjs/core';

class UploadData extends Component {

  render() {
    let data = this.props.result;

    return (
      <div>
        <div className="pt-dialog-body">

          <div className="row content-margin">
            <div className="col-sm-4">
              <label> Contract Address </label>
            </div>
            <div className="col-sm-8">
              <label> {data.contractAddress} </label>
            </div>
          </div>

          <div className="row content-margin">
            <div className="col-sm-4">
              <label> URI </label>
            </div>
            <div className="col-sm-8">
              <label> {data.uri} </label>
            </div>
          </div>

          <div className="row content-margin">
            <div className="col-sm-4">
              <label> Description </label>
            </div>
            <div className="col-sm-8">
              <label> {data.metadata} </label>
            </div>
          </div>

        </div>

        <div className="pt-dialog-footer">
          <div className="pt-dialog-footer-actions button-center">
            <Button
              intent={Intent.PRIMARY}
              onClick={() => this.props.closeModal()}
              text="Close"
            />
          </div>
        </div>
      </div>
    )
  }
}

export default UploadData;