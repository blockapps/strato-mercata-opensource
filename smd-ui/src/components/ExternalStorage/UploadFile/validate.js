import { isOauthEnabled } from "../../../lib/checkMode";

export function validate(values) {

  const errors = {};

  if (!isOauthEnabled() && !values.username) {
    errors.username = 'Select username';
  }

  if (!isOauthEnabled() && !values.address) {
    errors.address = 'Address required';
  }

  if (!isOauthEnabled() && !values.password) {
    errors.password = 'Password required';
  }

  if (!values.content) {
    errors.content = 'Attach file';
  }

  if (!values.provider) {
    errors.provider = 'Select provider';
  }

  if (!values.description) {
    errors.description = 'Attach file';
  }

  return errors;
}