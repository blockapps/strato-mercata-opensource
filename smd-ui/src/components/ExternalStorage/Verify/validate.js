export default function validate(values) {
  const errors = {};

  if (!values.contractAddress) {
    errors.contractAddress = "Can't be blank";
  }

  return errors;
}