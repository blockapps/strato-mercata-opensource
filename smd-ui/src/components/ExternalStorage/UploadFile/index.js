import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { Dialog } from '@blueprintjs/core';
import mixpanelWrapper from '../../../lib/mixpanelWrapper';
import { closeUploadModal, resetError } from './uploadFile.actions';
import { toasts } from '../../Toasts';
import { isOauthEnabled } from '../../../lib/checkMode';
import { fetchAccounts } from '../../Accounts/accounts.actions';
import UploadForm from './UploadForm';
import UploadData from './UploadData';

import './uploadFile.css';

class UplaodFile extends Component {

  componentWillReceiveProps(nextProps) {
    if (nextProps.uploadError) {
      toasts.show({ message: nextProps.uploadError });
      this.props.resetError();
    }
  }

  componentDidMount() {
    mixpanelWrapper.track("external_storage_loaded");
    !isOauthEnabled() && this.props.fetchAccounts(true, false);
  }

  render() {
    let result = this.props.result;

    return (
      <div>
        <Dialog
          isOpen={this.props.isOpen}
          onClose={() => {
            mixpanelWrapper.track('close_upload_modal');
            this.props.closeUploadModal();
          }}
          iconName={result ? 'saved' : 'pt-icon-upload'}
          title={result ? 'URI Upload Success' : 'Upload & attest'}
          className="pt-dark upload-dialog"
        >
          {result ? <UploadData result={this.props.result} closeModal={() => this.props.closeUploadModal()} /> : <UploadForm />}
        </Dialog>
      </div>
    )
  }
}

export function mapStateToProps(state) {
  return {
    isOpen: state.uploadFile.isOpen,
    uploadError: state.uploadFile.error,
    result: state.uploadFile.result
  };
}

const connected = connect(
  mapStateToProps,
  {
    closeUploadModal,
    resetError,
    fetchAccounts
  }
)(UplaodFile);

export default withRouter(connected);