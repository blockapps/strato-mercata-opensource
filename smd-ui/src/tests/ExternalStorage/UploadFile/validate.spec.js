import { validate } from '../../../components/ExternalStorage/UploadFile/validate';
import { mockFormData } from './mockUpload';

describe('UploadFile: validate', () => {

  test('with values', () => {
    expect(validate(mockFormData)).toMatchSnapshot();
  });

  test('with empty values', () => {
    const values = {};
    expect(validate(values)).toMatchSnapshot();
  });

});