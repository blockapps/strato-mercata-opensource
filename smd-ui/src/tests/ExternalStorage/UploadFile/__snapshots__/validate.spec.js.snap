// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadFile: validate with empty values 1`] = `
Object {
  "address": "Address required",
  "content": "Attach file",
  "description": "Attach file",
  "password": "Password required",
  "provider": "Select provider",
  "username": "Select username",
}
`;

exports[`UploadFile: validate with values 1`] = `Object {}`;
